ee/ucp/admin/configure/route-reflector-deployment.md

```diff
@@ -0,0 +1,210 @@
+                 valueFrom:
+                   configMapKeyRef:
+                     name: calico-config
+                     key: etcd_key # Location of the client certificate for etcd.
+               - name: ETCD_CERT_FILE
+                 valueFrom:
+                   configMapKeyRef:
+                     name: calico-config
+                     key: etcd_cert
+               - name: IP
+                 valueFrom:
+                   fieldRef:
+                     fieldPath: status.podIP
+             volumeMounts:
+               - mountPath: /calico-secrets
+                 name: etcd-certs
+             securityContext:
+               privileged: true
+         nodeSelector:
+           com.docker.ucp.kubernetes.calico/route-reflector: "true"
+         volumes:
+         # Mount in the etcd TLS secrets.
+           - name: etcd-certs
+             secret:
+               secretName: calico-etcd-secrets
+  ```
+  kubectl create -f calico-rr.yaml
+  ```
+  
+6) Use ```calicoctl``` to get the current bgp configuration and modify it to turn off node to node bgp mesh.
+
+   ```calicoctl get bgpconfig -o yaml > bgp.yaml```
+   
+   Set the nodeToNodeMeshEnabled to false in the spec.
+  
+   ``` bgp.yaml:```
+   ```
+   ....
+   spec:
+     asNumber: 63400
+     logSeverityScreen: Info
+     nodeToNodeMeshEnabled: false
+   ....
+   ```
+   Replace the bgp configuation with the modified ```bgp.yaml```
+   
+   ```
+   calicoctl replace -f bgp.yaml
+   ```
+   
+   If the bgpConfig object is empty, create the bgp config object.
+   ```
+   cat << EOF | calicoctl create -f -
+   apiVersion: projectcalico.org/v3
+   kind: BGPConfiguration
+   metadata:
+     name: default
+   spec:
+     logSeverityScreen: Info
+     nodeToNodeMeshEnabled: false
+     asNumber: 63400
+   EOF
+   ```
+   
+7) Create a bgp RR peer configuration. You can create multiple configuration for RRs
+
+   ```
+   calicoctl create -f - << EOF
+   apiVersion: projectcalico.org/v3
+   kind: BGPPeer
+   metadata:
+     name: bgppeer-global
+   spec:
+     peerIP: <IP_RR> #This will be the IP of node where the calico route reflector pod will be deployed.
+     asNumber: <AS_NUM> #Use the same asNumber from the configuration above step 6).
+   EOF
+   ```
+   The bgp RR configuration needs to be added for every Route Reflector that will be deployed.
+   Refer to https://docs.projectcalico.org/v3.1/usage/routereflector/calico-routereflector for more information.
+
+
+8) If you have calico-node pod running on the nodes marked for route reflector in step 3). You will need to manually delete the calico-node pod. We recommend this to avoid running both calico route reflector and calico-node pods on the same node.
+   
+   ```
+   kubectl get pods -n kube-system -o wide | grep ubuntu-0
+   calico-node-t4lwt                        2/2       Running   0          3h        172.31.20.89     ubuntu-0
+   
+   kubectl delete pod -n kube-system calico-node-t4lwt
```

Comment:
Will the `ucp-agent` or any other UCP component redeploy this pod, once I manually delete it?

Reply:
no since we have tainted the nodes in the previous steps, it will not be